core/domain/exp_domain.py

```diff
@@ -3030,91 +3112,1490 @@ def _convert_states_v51_dict_to_v52_dict(
+                if answer_group['outcome']['dest'] == state_name:
+                    answer_group['outcome']['labelled_as_correct'] = False
+
+            state_dict['interaction']['answer_groups'] = answer_groups
 
-            for answer_group in interaction['answer_groups']:
-                content_id_list.append(
-                    answer_group['outcome']['feedback']['content_id']
+        return states_dict
+
+    @classmethod
+    def _choices_should_be_unique_and_non_empty(
+        cls, choices: List[str],
+        answer_groups: List[state_domain.AnswerGroup],
+        is_item_selection_interaction = False
+    ) -> Tuple(Any):
+        """Handles choices present in the ItemSelectionInput or
+        in MultipleChoiceInput interactions, implements the following -
+            - If only one choice is empty then simply removes it
+            - If multiple choices are empty replace them with `Choice 1` ,
+            `Choice 2` etc
+            - If choices are duplicate, removes the later choice
+            - Remove the rules whose choices has been deleted
+
+        Args:
+            choices: List[str]. A list of choices.
+            answer_groups: List[state_domain.AnswerGroup]. The list of answer
+            groups.
+            is_item_selection_interaction: bool. If the answer group belongs
+            to ItemSelectionInput interaction or not.
+
+        Returns:
+            choices_to_save: List[str]. The list of valid choices.
+        """
+        empty_choices = []
+        seen_choices = []
+        choices_to_remove = []
+        invalid_choices_index = []
+        invalid_choices_content_ids = []
+        for choice in choices:
+            if choice['html'].strip() in ('<p></p>', ''):
+                empty_choices.append(choice)
+
+        # Only one choice is empty.
+        if len(empty_choices) == 1:
+            invalid_choices_index.append(choices.index(empty_choices[0]))
+            invalid_choices_content_ids.append(empty_choices[0]['content_id'])
+            choices.remove(empty_choices[0])
+        # Multiple choices are empty.
+        else:
+            for idx, empty_choice in enumerate(empty_choices):
+                empty_choice['html'] = (
+                    '<p>' + 'Choice ' + str(idx+1) + '</p>'
                 )
 
+        # Duplicate choices.
+        for choice in choices:
+            if choice['html'] not in seen_choices:
+                seen_choices.append(choice['html'])
+            else:
+                choices_to_remove.append(choice)
+                invalid_choices_index.append(choices.index(choice))
+                invalid_choices_content_ids.append(choice['content_id'])
+
+        for choice_to_remove in choices_to_remove:
+            choices.remove(choice_to_remove)
+
+        # Remove rules of ItemSelection interaction whose choice
+        # has been deleted.
+        if is_item_selection_interaction:
```

Comment:
you can easily combine this block of code for item selection with the one for multiple choice input, right? They only differ in the `for rule_spec in answer_group['rule_specs']:` block

Reply:
Yep! Updated